### ISSUE TITLE
Required option on a group main command makes it hard to get help on subcommands

### ISSUE THREAD

**Issue by haridsv** [pallets/click#2020]
When grouping commands, `--help` can be used on both the main and subcommands. If the main command defines a `required` option, then invoking help on subcommands  becomes harder, as click requires passing the required options. E.g., see the below slight modification of the [sample](https://click.palletsprojects.com/en/8.0.x/quickstart/#nesting-commands) from documentation

```
import click

@click.group()
@click.option("--db-path", required=True)
def cli(db_path):
    pass

@cli.command()
def initdb():
    click.echo('Initialized the database')

@cli.command()
def dropdb():
    click.echo('Dropped the database')

if __name__ == '__main__':
    cli()
```

If you save the above as `/tmp/t.py` and invoke it as `python /tmp/t.py initdb --help`, you would get:

```
Usage: t.py [OPTIONS] COMMAND [ARGS]...
Try 't.py --help' for help.

Error: Missing option '--db-path'.
```

To be able to get the help for `initdb`, you would be forced to specify a value for the required option, something like `python /tmp/t.py --db-path /tmp/t initdb --help`

```
Usage: t.py initdb [OPTIONS]

Options:
  --help  Show this message and exit.
```

Imaging the problem if the main command were to have multiple `required` options.

*Expected behavior*: One should be able to invoke `--help` for a subcommand without having to specify any required params for the main command.

Environment:

- Python version: `3.9.1`
- Click version: `8.0.1`

**Comment by davidism** [comment]
See existing issues about this. This is not possible due to the fact that Click commands and parameters can be completely dynamic.

**Comment by haridsv** [comment]
@davidism Apologies for not searching through closed issues. I only searched the open issues as I was so sure that there would be an open issue for it. I went through closed issues and see #814 and #295 and more details on why this won't be fixed.

**Comment by haridsv** [comment]
I ended up adding an explicit check instead of relying on `required`, something like this:

```
import click

def handle_required_opts(group, *opts):
    for opt in opts:
        if not opt in group.params:
            click_opt = next(o for o in group.params if o.name == opt)
            raise click.UsageError(f"Error: Missing option '{click_opt.opts[0]}'")

@click.group()
@click.option("--db-path", help="(required)")
def cli(db_path):
    pass

@cli.command()
def initdb():
    handle_required_opts(cli, 'db_path')
    click.echo('Initialized the database')

@cli.command()
def dropdb():
    handle_required_opts(cli, 'db_path')
    click.echo('Dropped the database')

if __name__ == '__main__':
    cli()
```

I was hoping to make use of `click.get_current_context().command` to figure out the current command and then reach the group, but command doesn't seem to have a reference to its group (only the other way).

@davidism Do you have any suggestion to improve it? Can this be converted to a decorator?
